app/models/piece.rb

```diff
@@ -66,4 +66,24 @@ def is_move_blocked(destination_x, destination_y)
       return is_diagonal_move_blocked(destination_x, destination_y)
     end
   end
+
+  # IS THERE A PIECE AT THIS SPACE OF THE SAME COLOR AS SELF
+
+  def not_occupied_by_me?(destination_x, destination_y)
+    game.pieces.where(piece_color: self.piece_color, current_x: destination_x, current_y: destination_y).empty?
+  end
+
+  # MOVE MUST BE ON THE BOARD
+
+  def on_the_board?(destination_x, destination_y)
+    destination_x.between?(0, 7) && destination_y.between?(0, 7)
+  end
+
+  # MOVE METHOD
+
+  def move?(destination_x, destination_y)
+    not_occupied_by_me?(destination_x, destination_y) &&
+    on_the_board?(destination_x, destination_y) &&
+    valid_move?(destination_x, destination_y)
+  end
```

Comment:
You're correct. When I was testing I thought it let me move off the board. I didn't realize the same logic is in `valid_move?`. I've also placed constraints with the JavaScript on moves, it will not let you move outside of the `chessboard`. So, we have checks in multiple places. I'll remove the `on_the_board?` method and push to the branch.